java-checks/src/main/java/org/sonar/java/checks/SubClassStaticReferenceCheck.java

```diff
@@ -99,7 +99,7 @@ public void visitClass(ClassTree tree) {
     @Override
     public void visitIdentifier(IdentifierTree tree) {
       Type type = tree.symbolType();
-      if (!classType.equals(type) && type.isSubtypeOf(classType)) {
+      if (!classType.is(type.fullyQualifiedName()) && type.isSubtypeOf(classType)) {
```

Comment:
This bothers me as it sounds rather like working around the API rather than adressing the real issue. As you tried to improve things with method type, then the is method should most probably be improved to handle those cases... But I do wonder if we actually really want "is" to answer true when we compare classtype and methodtype. 

IMO best course of action would be to have an explicit instanceof here. As you dealt with in other checks